### PR TITLE
fix: navigation spacing [NP-816]

### PR DESCRIPTION
**Pull request by baseonmars**
* Site Navigation buttons no longer introduce excess spacing when tabbing
* Site Navigation no longer outline links in blue causing a confusing visual effect which does not align with Figma design
* Added visual regression tests so that appearance of Site Navigation button active/focus state. 
** note: tests are currently not run as part of CI due to instability.

**Before (THE OLD VERSION, YUK!)**

menu-before

**After (the new version, much preferred)**

menu-after

### CODE REVIEW

File: testing/visual-regression/backstop.json [hunk 1]
```diff
@@ -403,6 +403,54 @@
...
+      "requireSameDimensions": true,
+      "viewports": [
+        {
+          "label": "phone",
+          "width": 320,
+          "height": 400
+        },
+        {
+          "label": "Small",
+          "width": 641,
```
Comment (luke-hill) [inline testing/visual-regression/backstop.json:439]: are these numbers correct? It feels weird that we've gone from 320 x 400 to 641 x 200.

The ones below look weird too. But maybe only the width matters. I'm not particularly clued up on this area, so disregard if not relevant.

**Review by luke-hill — COMMENTED** [review]
I had a play with this offline, and found the new tabbing to much cleaner and easier to use.

You mention hover states have changed, but they looked the same for me when I was mousing over the menu items (Or the difference was very small).

File: testing/visual-regression/backstop.json (hunk 1 above)
Comment (baseonmars) [reply testing/visual-regression/backstop.json:439]: I coppied these from the other navigation test, I can't comment either way. If I had to guess it's related to the triggering the greedy nav  behaviour.

**Review by davidsauntson — CHANGES_REQUESTED** [review]
Pls can you update the changelog 🥥

**Review by soniaturcotte — APPROVED** [review]

**Review by davidsauntson — APPROVED** [review]
🦞 noice

**Pull request merged**